### PR TITLE
Update Extensions Configuration Secrets and Identity sdk from t1 to t2 

### PR DESCRIPTION
**Pull request by Tianyun-G**
1.Update Extensions Configuration Secrets and Identity sdk from t1 to t2 

2.Update files:

- i. src/inventory-service/InventoryService.Api/InventoryService.Api.csproj
- ii. src/inventory-service/InventoryService.Api/Program.cs
- iii. src/inventory-service/README.md

@scottaddie,@AlexGhiondea,@schaabs,@pakrym and @christothes for notification.

### CODE REVIEW

File: src/inventory-service/InventoryService.Api/Program.cs [hunk 1]
```diff
@@ -24,15 +24,12 @@ public static void Main(string[] args)
...
+                    var keyVaultUrl = config["KeyVaultUrl"];
+                    if (!string.IsNullOrEmpty(keyVaultUrl))
                     {
-                        var azureServiceTokenProvider = new AzureServiceTokenProvider();
-                        var keyVaultClient = new KeyVaultClient(
-                            new KeyVaultClient.AuthenticationCallback(
-                                azureServiceTokenProvider.KeyVaultTokenCallback));
-                        builder.AddAzureKeyVault(
-                            keyVaultEndpoint, keyVaultClient, new DefaultKeyVaultSecretManager());
+                        var secretClient = new SecretClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
```
Comment (pakrym) [inline src/inventory-service/InventoryService.Api/Program.cs]:
```suggestion
```

**Comment by v-xuto** [comment]
@pakrym   We have updated by your comments, please re-review this PR.

**Comment by ahsonkhan** [comment]
I don't work on the .NET Identity SDKs, so I don't think I am the appropriate person to review this. Maybe it was meant for someone else :)

For C++ SDK, we have @antkmsft for Identity.

**Comment by v-xuto** [comment]
@pakrym We have updated by your comments, please re-review this PR.

**Review by pakrym — APPROVED** [review]

**Pull request merged**